apps/testingaccessibility/src/components/team/self-redeem-button.tsx

```diff
@@ -25,8 +26,23 @@ const SelfRedeemButton: React.FC<
       onClick={() => {
         if (userEmail) {
           setIsLoading(true)
-          handleSelfRedeem(userEmail, bulkCouponId, (redeemedPurchase) => {
-            onSuccess(redeemedPurchase)
+          handleSelfRedeem(userEmail, bulkCouponId, (params) => {
+            if (params.status === 'success') {
+              onSuccess(params.redeemedPurchase)
+            } else {
+              // TODO: report to sentry or support?
+              console.log(params.error)
+              if (params.error.startsWith('already-purchased-')) {
+                toast.error(
+                  'You have already redeemed a seat for yourself. Please contact support if you are having trouble accessing it.',
+                )
+              } else {
+                toast.error(
+                  'We were unable to redeem a seat for this account. If the issue persists, please reach out to support.',
+                )
+              }
+            }
+
             setIsLoading(false)
```

Comment:
If successful, call the `onSuccess` handler.

If failed, then surface some info to the user so that they know what is going on.

Lastly, always setting loading back to `false` so that the spinner doesn't sit there indefinitely.